<a href="https://colab.research.google.com/github/KshitijPriyank/Python/blob/main/26_8_2020_KSHITIJ_PROJECT_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

***ACTIVITY 1:-***

In [ ]:
#Activity 1: Multiply a number with the previous number
var=input('enter a number')
var=int(var)
def multiplayer(var):
  if var<0:
    print('invalid number')
    var=input('enter a number')
  elif var==0:
    print('your answer is 1')
  else:
    mult=1
    while var>0:      #3 true
      product=mult*var   #3*1
      mult=product
      var=var-1         #2
    return product
multiplayer(var)









enter a number5


120

***ACTIVITY 2:-***

In [ ]:
#Activity 2: Multiplication table
inputnum=input(' enter a number ')    #2
num=1
inputnum=int(inputnum)
print(inputnum,'x',num,'=',inputnum) #2x1=2
while num<10:     #1<10,2<10
  num=num+1       #2,3
  product=inputnum*num   #2x2=4, 2x3=6
  print(inputnum,'x',num,'=',product)#2x2=4,2x3=6



 enter a number 3
3 x 1 = 3
3 x 2 = 6
3 x 3 = 9
3 x 4 = 12
3 x 5 = 15
3 x 6 = 18
3 x 7 = 21
3 x 8 = 24
3 x 9 = 27
3 x 10 = 30


### Instructions

#### Goal of the Project

This project is designed for you to practice and solve the activities that are based on the concepts covered in the following lessons:

 1.  While Loop, Data-Type Conversion and Conditional Statements.

 2.  Improving The Algorithm


---

#### Getting Started:

1. Click on this link to open the Colab file for this project.

    https://colab.research.google.com/drive/17lt4EdjsvfOIys__BwY0iOIrGOCIjhgg?usp=sharing

2. Create a duplicate copy of the Colab file as described below.

  - Click on the **File menu**. A new drop-down list will appear.

   <img src='https://student-datasets-bucket.s3.ap-south-1.amazonaws.com/images/lesson-0/0_file_menu.png' width=500>

  - Click on the **Save a copy in Drive** option. A duplicate copy will get created. It will open up in the new tab on your web browser.

  <img src='https://student-datasets-bucket.s3.ap-south-1.amazonaws.com/images/lesson-0/1_create_colab_duplicate_copy.png' width=500>

3. After creating the duplicate copy of the notebook, please rename it in the **YYYY-MM-DD_StudentName_Project3** format.

4. Now, write your code in the prescribed code cells.


---

### Activities

#### Activity 1: Multiply a number with the previous number

Write a function to print the result of a number multiplied with its previous number until the previous number is 1.

For example:
*    If the number is 5, then result = 5 * 4 * 3 * 2 * 1 = 120
*    If the number is 8, then result = 8 * 7 * 6 * 5 * 4 * 3 * 2 * 1 = 40320

Follow the steps given below to achieve the desired result:

  - **Step 1:** Ask the user to enter a number. Convert the data type of the user input to `integer` and then store it in a variable `num`.

  - **Step 2:** Create a function `multiplier()` that has `n` as the parameter i.e. `multiplier(n)`. Here, `n` is used to hold user input within this function.  

  - **Step 3:** Inside this function, check whether the user has entered a positive number or not using an `if-elif-else` statement. If the user enters a negative number (i.e. `n<0`), then print the message `Sorry, the number cannot be negative`. Else, if the user enters `0` (i.e. `n=0`), then print the message `The result is 1`.

  - **Step 4:** Else, if the user enters a positive number (i.e. `n>0`) then:

      1. Initialize a variable `mult = 1`.

      2. Keep on multiplying the value stored in variable `mult` with value stored in variable `n` (`mult * n`) till the value of `n` is greater than 0.

      3. Decrement the value of `n` on each execution of the `while` loop.

  - **Step 5:** As soon as the condition of the `while` loop becomes false, we get our desired result in the variable `mult`. Print the `mult` variable. This completes the implementation of our `multiplier()` function.

  - **Step 6:** Now call the `multiplier()` function and pass the variable `num` (a variable which carries user input) as a parameter to this function.

In [ ]:
# Write your code here
# Step 1: Accept user input.

# Step 2: Define a function 'multiplier(n)' .

      # Step 3: Inside the function, check if the number is negative, positive or 0

      # If the number is positive, perform Step 4 and Step 5

# Step 6: Call the 'multiplier()' function to execute


---

#### Activity 2: Multiplication table

Take input from the user and display the multiplication table.

For example: if the number entered is 12 then expected output is:

  12 x 1 = 12

  12 x 2 = 24

  ....
  
  12 x 10 = 120

Follow the steps given below to achieve the desired result:

- **Step 1:** Ask the user to enter a number. Convert the data type of user input to integer and then store it in a variable `num`.

- **Step 2:** Initialize a variable `i=1` (This variable is used to make sure that the `while` loop is executed only upto 10, starting from 1).

- **Step 3:** Create a `while` loop. This loop will be executed 10 times. So, the condition for `while` loop will be `i<=10`.

- **Step 4:** Inside this loop, multiply the `num` variable with `i` variable and print its result. Also increment the value of `i` by 1 inside this loop. The execution stops once the condition for `while` loop becomes invalid.


In [ ]:
# Write your code here
# Step 1: Get the user input

# Step 2: Initialize variable 'i'

# Step 3: Execute a 'while' loop 10 times

  # Step 4: Multiply the 'num' variable with 'i' variable and print its result. Increment the value of 'i' by 1

---

### Submitting the Project:

1. After finishing the project, click on the **Share** button on the top right corner of the notebook. A new dialog box will appear.

  <img src='https://student-datasets-bucket.s3.ap-south-1.amazonaws.com/images/project-share-images/2_share_button.png' width=500>

2. In the dialog box, make sure that '**Anyone on the Internet with this link can view**' option is selected and then click on the **Copy link** button.

   <img src='https://student-datasets-bucket.s3.ap-south-1.amazonaws.com/images/project-share-images/3_copy_link.png' width=500>

3. The link of the duplicate copy (named as **YYYY-MM-DD_StudentName_Project3**) of the notebook will get copied

   <img src='https://student-datasets-bucket.s3.ap-south-1.amazonaws.com/images/project-share-images/4_copy_link_confirmation.png' width=500>

4. Go to your dashboard and click on the **My Projects** option.
   
   <img src='https://student-datasets-bucket.s3.ap-south-1.amazonaws.com/images/project-share-images/5_student_dashboard.png' width=800>

  <img src='https://student-datasets-bucket.s3.ap-south-1.amazonaws.com/images/project-share-images/6_my_projects.png' width=800>

5. Click on the **View Project** button for the project you want to submit.

   <img src='https://student-datasets-bucket.s3.ap-south-1.amazonaws.com/images/project-share-images/7_view_project.png' width=800>

6. Click on the **Submit Project Here** button.

   <img src='https://student-datasets-bucket.s3.ap-south-1.amazonaws.com/images/project-share-images/8_submit_project.png' width=800>

7. Paste the link to the project file named as **YYYY-MM-DD_StudentName_Project3** in the URL box and then click on the **Submit** button.

   <img src='https://student-datasets-bucket.s3.ap-south-1.amazonaws.com/images/project-share-images/9_enter_project_url.png' width=800>

---